src/Compilers/Core/Portable/AssemblyUtilities.cs

```diff
@@ -63,7 +63,7 @@ public static ImmutableArray<string> FindAssemblySet(string filePath)
                         var reference = metadataReader.GetAssemblyReference(handle);
                         var referenceName = metadataReader.GetString(reference.Name);
 
-                        string referencePath = Path.Combine(directory, referenceName + ".dll");
+                        string referencePath = Path.Combine(directory!, referenceName + ".dll");
```

Comment:
How do we know that `directory` is not-null? `Path.GetDirectoryName` could return `null` for a root directory, I think

Reply:
In this case `directory` comes from a file path. If it weren't a file path then the call to `FileUtilities.OpenRead(assemblyPath)` above would fail. Hence we have sufficient guarantees that `directory` will be non-null in this case (at least to my understanding).